reconcile/openshift_saas_deploy_trigger_cleaner.py

```diff
@@ -0,0 +1,71 @@
+import logging
+
+from datetime import datetime, timedelta
+from dateutil import parser
+
+import reconcile.queries as queries
+
+from reconcile.utils.oc import OC_Map
+from reconcile.utils.defer import defer
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.saasherder import Providers
+
+
+QONTRACT_INTEGRATION = 'openshift-saas-deploy-trigger-cleaner'
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+
+def within_retention_days(resource, days):
+    metadata = resource['metadata']
+    creation_date = parser.parse(metadata['creationTimestamp'])
+    interval = datetime.utcnow().date() - creation_date.date()
+
+    return interval < timedelta(days=days)
+
+
+@defer
+def run(dry_run, thread_pool_size=10, internal=None,
+        use_jump_host=True, defer=None):
+    settings = queries.get_app_interface_settings()
+    pipelines_providers = queries.get_pipelines_providers()
+    tkn_namespaces = [pp['namespace'] for pp in pipelines_providers
+                      if pp['provider'] == Providers.TEKTON]
+
+    oc_map = OC_Map(
+        namespaces=tkn_namespaces,
+        integration=QONTRACT_INTEGRATION,
+        settings=settings, internal=internal,
+        use_jump_host=use_jump_host,
+        thread_pool_size=thread_pool_size)
+    defer(lambda: oc_map.cleanup())
+
+    for pp in pipelines_providers:
+        retention = pp.get('retention')
+        if not retention:
+            continue
```

Comment:
Should we hava default retention setting instead of ignoring those pipelines?

Reply:
good next iteration? for now we have very few namespaces that have A LOT of PipelineRuns.

Reply:
Next iteration wfm

Reply:
+1